presto-main/src/main/java/com/facebook/presto/operator/scalar/ArrayIntersectFunction.java

```diff
@@ -52,4 +54,22 @@ public static Block intersect(
 
         return typedSet.getBlock();
     }
+
+    @SqlInvokedScalarFunction(value = "array_intersect", deterministic = true, calledOnNullInput = false)
```

Comment:
This needs to be in a separate class if I recall correctly, otherwise function parsing would fail. Function annotation parsing does not support a mix of @ScalarFunction and @SqlInvokedScalarFunction.

Reply:
Ya i fixed that. I think it works now. At least `TpchQueryRunner` can start and all 3 versions of the functions are listed, and `TestArrayIntersectFunction` pass.